meson.build

```diff
@@ -0,0 +1,40 @@
+project('my_module', 'cpp',
+        default_options: ['cpp_std=c++17'])
+
+# ===========================================================
+# Python 3
+# ===========================================================
+
+# python 3 dependency
+python_dep = dependency('python3')
+
+# python 3 import for extension_module
+python_module = import('python')
+inst = python_module.find_installation(get_option('python_executable'))
```

Comment:
For consistency here, your python_dep should probably be aligned with the found installation (which you want to install as). You can guarantee this by using [`inst.dependency()`](https://mesonbuild.com/Python-module.html#dependency).

Reply:
Yep, That was giving me issues building on different python versions. It's switched in the latest commit. I'm still having problems building on windows tho